File: src/API/Microsservices/Marketplace/Sonorus.Marketplace.Application/Commands/CreateProduct/CreateProductInputModel.cs
using Microsoft.AspNetCore.Http;
using Sonorus.Marketplace.Core.Enums;

namespace Sonorus.Marketplace.Application.Commands.CreateProduct;

public class CreateProductInputModel {
    public string Name { get; set; } = null!;
    public string? Description { get; set; }
    public decimal Price { get; set; }
    public ConditionType Condition { get; set; }
    public IEnumerable<IFormFile> Medias { get; set; } = [];
}

File: src/app/lib/src/dtos/input_models/create_product_input_model.dart
import "package:dio/dio.dart";
import "package:image_picker/image_picker.dart";

import "package:sonorus/src/domain/enums/condition_type.dart";

class CreateProductInputModel {
  final String name;
  final double price;
  final String? description;
  final ConditionType condition;
  final List<XFile> medias;

  CreateProductInputModel({
    required this.name,
    required this.price,
    this.description,
    required this.condition,
    required this.medias
  });

  FormData toFormData() {
    final FormData formData = FormData.fromMap({
      "name": this.name,
      "price": this.price,
      "description": this.description,
      "condition": this.condition.id
    });
    formData.files.addAll(medias.map((media) => MapEntry("medias", MultipartFile.fromFileSync(media.path))));
    return formData;
  }
}